src/main/java/bio/terra/workspace/service/workspace/WorkspaceService.java

```diff
@@ -126,6 +127,22 @@ public Workspace getWorkspace(UUID id, AuthenticatedUserRequest userReq) {
     return validateWorkspaceAndAction(userReq, id, SamConstants.SAM_WORKSPACE_READ_ACTION);
   }
 
+  /**
+   * Update an existing workspace. Currently, can change the workspace's display name or description.
+   * @param workspaceId workspace of interest
+   * @param name name to change - may be null
+   * @param description description to change - may be null
+   */
+  public Workspace updateWorkspace(
+      AuthenticatedUserRequest userReq,
+      UUID workspaceId,
+      @Nullable String name,
+      @Nullable String description) {
+    validateWorkspaceAndAction(userReq, workspaceId, SamConstants.SAM_WORKSPACE_WRITE_ACTION);
```

Comment:
I think we want to get the workspace and report any 404 before we do the validate. This is an extra step, but is nicer to the caller and is the convention.
@zloery check me on this, please

Reply:
I am not sure I understand... This is the function that getWorkspace calls?

Reply:
Nevermind The validate call does the right thing.